docs/plugins.md

```diff
@@ -0,0 +1,267 @@
+## Plugin Support
+
+You can customize the linter with rules that are more specific to your use case with the `ember-template-lint` plugin system.
+
+Plugins can define new rules and set up default configurations that can be extended.
+
+### Adding Plugins to Your Configuration
+
+In order to enable a plugin, you must add it to the `plugins` key in your configuration file.
+
+`plugins` accepts an array of either objects, or strings that resolve to files that export plugin objects.
+
+#### Define a plugin inline
+
+```javascript
+// .template-lintrc.js
+
+module.exports = {
+  plugins: [
+    {
```

Comment:
The doc currently explains inline plugins differently from the types of plugins that you would require, and they are not really any different.  Each entry in the plugins list must conform to a general "interface" (in a general hand wavy sense). A "Plugin" is a specific thing, that has an expected set of properties.

E.g. it has a `name`, `rules` (Object whose keys are the rule names and the values are the rules), `configurations` (Object whose keys are the config names, and the values are the config values), etc.

If we can define what a "Plugin" is, then the sections for `Define a plugin inline` and `Define a plugin based on a relative path` are really no different.